Spotify Beta: Add album image type to nowPlayingMenuBar icon

kvdo2 opened the pull request:
## Description
I added two options to the nowPlayingMenuBar icons:
1. Spotify icon
2. Track album image.

## Screencast

<img width="306" alt="image" src="https://user-images.githubusercontent.com/78251524/232950129-4e28aca8-5d2b-4658-9d37-13c12aafaf71.png">


<img width="270" alt="image" src="https://user-images.githubusercontent.com/78251524/232952884-7ad936bc-e560-4999-846f-dc7e138c03b9.png">


## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

peduarte commented on extensions/spotify-beta/src/nowPlayingMenuBar.tsx:
```diff
@@ -34,8 +35,17 @@ import { useCachedState } from "@raycast/utils";
+  const preferences = getPreferenceValues<{
+    maxTextLength?: boolean;
+    showEllipsis?: boolean;
+    iconType?: NowPlayingMenuBarIconType;
```
Can we use string union instead of enum please? It's what I've been using throughout the codebase. Thanks!

peduarte replied:
```suggestion
    iconType?: "spotify-icon" | "album-cover";
```

peduarte commented on extensions/spotify-beta/package.json:
```diff
@@ -63,6 +63,24 @@
+          "data": [
+            {
+              "title": "Spotify Icon",
+              "value": "spotify"
```
Can we call this value `spotify-icon` to be more verbose? Thanks!

peduarte replied:
```suggestion
              "value": "spotify-icon"
```

peduarte commented on extensions/spotify-beta/package.json:
```diff
@@ -63,6 +63,24 @@
+            },
+            {
+              "title": "Album Image",
+              "value": "album"
```
Can we call this `album-cover` please? Thanks!

```suggestion
              "value": "album-cover"
```

peduarte commented on extensions/spotify-beta/package.json:
```diff
@@ -63,6 +63,24 @@
+              "value": "spotify"
+            },
+            {
+              "title": "Album Image",
```
Using "Album Cover" to be more verbose

```suggestion
              "title": "Album Cover Image",
```

peduarte commented on extensions/spotify-beta/package.json:
```diff
@@ -63,6 +63,24 @@
+        },
+        {
+          "name": "iconType",
+          "title": "Menu Bar Icon type",
```
Since these preferences are already nested in the Menu Bar command, I don't think we need to include in the title

I'd move this to be the first preference as well

```suggestion
          "title": "Icon type",
```

peduarte commented on extensions/spotify-beta/package.json:
```diff
@@ -63,6 +63,24 @@
+        {
+          "name": "iconType",
+          "title": "Menu Bar Icon type",
+          "description": "Choose between the default Spotify icon, or the album image of the current track",
```
```suggestion
          "description": "Choose between the default Spotify icon, or the album cover image of the current track",
```

peduarte commented on extensions/spotify-beta/src/nowPlayingMenuBar.tsx:
```diff
@@ -163,7 +176,14 @@ function NowPlayingMenuBarCommand({ launchType }: LaunchProps) {
       isLoading={currentlyPlayingIsLoading || currentlyPlayingUriIsLoading || playbackStateIsLoading}
-      icon={{ source: { dark: "menu-icon-dark.svg", light: "menu-icon-light.svg" } }}
+      icon={
+        preferences.iconType === NowPlayingMenuBarIconType.Album && albumImageUrl
```
```suggestion
        preferences.iconType === "album-cover" && albumImageUrl
```

peduarte reviewed (changes requested):
Thanks for your contribution. This is something @thomaspaulmann had also mentioned :) 

I'd like to try and keep the preferences to a minimum, but seems like this is something a few people want to be able to do, so let's go for it

thomaspaulmann commented:
Oh that's neat. The lil album cover looks lovely in the menu bar! @peduarte the setting could be extended for the folks that want to have the colored version. So you could have three options: Monochrome Icon, Green Icon, Album Cover.

BTW is it always the album cover? Or what do you show when you play the radio of an artist?

kvdo2 commented:
@thomaspaulmann  Now it supports displaying the cover image of songs and episodes.

@peduarte Thank you for your review. I have changed the relevant code. After testing, I can successfully get the cover image of episode, So I changed the option title  `Album Cover Image` to `Cover Image`. Because it is inclusive of all types of cover images, not just for album covers of songs.

peduarte reviewed (approved):
Nice one, thanks for the changes. LGTM, let's take it for a spin 😎

kvdo2 commented:
I added cover icon support for the playlist in`Add to Playlist` an in the menu bar. For a playlist without a cover, I really can't find a more suitable icon than `Icon.Music`, even though it repeats with `Start Radio`. This may need to be reconsidered.

This is a example.
<img width="296" alt="image" src="https://user-images.githubusercontent.com/78251524/233390315-2ea49b58-f1dc-46fa-83e6-dd24d4bff465.png">

peduarte commented:
I think let's keep this PR just focused on cover image for menu bar. Not sure how I feel about cover image for the list, might be too much going on

peduarte commented:
Happy to merge once conflict's fixed. Thanks!

kvdo2 commented:
> Happy to merge once conflict's fixed. Thanks!

The conflict has been fixed, but I am very confused about this code, why is it written in this way?The two variable names in the OR statement are the same.

`isLoading={currentlyPlayingIsLoading || currentlyPlayingIsLoading}`

peduarte commented:
@kvdo2 oh, that's a mistake 😅 I'll merge this and fix it
